pkg/registry/pod/strategy.go

```diff
@@ -153,6 +154,23 @@ func (podStatusStrategy) ValidateUpdate(ctx api.Context, obj, old runtime.Object
 	return validation.ValidatePodStatusUpdate(obj.(*api.Pod), old.(*api.Pod))
 }
 
+func CheckPod(r *registry.Store, ctx api.Context, err error, obj runtime.Object) error {
+	if !errors.IsAlreadyExists(err) {
+		return err
+	}
+	pod := obj.(*api.Pod)
+	retObj, getPodErr := r.Get(ctx, pod.Name)
+	if getPodErr != nil {
+		return err
+	}
+	retPod := retObj.(*api.Pod)
+	if api.IsPodTerminated(retPod) {
+		status_err, _ := err.(*errors.StatusError)
+		status_err.ErrStatus.Message += fmt.Sprintf("\n Existant pod is in terminated state! Please check its status and delete it!\n Use `kubectl get pods/%s` \n", pod.Name)
```

Comment:
I added general check for graceful deletion state, in which case the error would be the following:

`Error from server: error when creating "../podConfigmap.yaml": pods "dapi-test-pod1" already exists- the object may be in the process of deletion!`

And it could be do in generic way as this field is placed in Metadata structure which is common for all objects and has helpers using reflect to reach fields in it for all objects. 

But if we want to check perform check based on object-specific fields (like pod status) we can't do it in the same way for all objects and we have already mechanism to implement object-specific logic using assigning registry.Store struct funcs and add one more generic function which is called in case of Create resource failure if it is defined.